src/main/java/org/littleshoot/proxy/impl/ProxyToServerConnection.java

```diff
@@ -954,13 +954,6 @@ void connectionSucceeded(boolean shouldForwardInitialRequest) {
         } else {
             LOG.debug("Dropping initial request: {}", initialRequest);
         }
-
-        // we're now done with the initialRequest: it's either been forwarded to the upstream server (HTTP requests), or
-        // completely dropped (HTTPS CONNECTs). if the initialRequest is reference counted (typically because the HttpObjectAggregator is in
-        // the pipeline to generate FullHttpRequests), we need to manually release it to avoid a memory leak.
-        if (initialRequest instanceof ReferenceCounted) {
-            ((ReferenceCounted)initialRequest).release();
-        }
```

Comment:
This `release` call is not required as the message is deallocated here 
https://github.com/barchart/barchart-project-netty/blob/master/transport/src/main/java/io/netty/channel/SimpleChannelInboundHandler.java#L110. It is done automatically by netty by default. This is way we had https://github.com/adamfisk/LittleProxy/issues/357.

There is also `retain` call here https://github.com/adamfisk/LittleProxy/blob/70edbe1c502d79449fb5ceb9d00cca6a0759d56f/src/main/java/org/littleshoot/proxy/impl/ProxyToServerConnection.java#L318. By default netty releases a message after it is written to upstream and as we still need that message in client to proxy connection we need to retain it so we can release it here https://github.com/barchart/barchart-project-netty/blob/master/transport/src/main/java/io/netty/channel/SimpleChannelInboundHandler.java#L110